src/main/app/fact/IdamApi.ts

```diff
@@ -0,0 +1,37 @@
+import config from 'config';
+import {AxiosError, AxiosInstance} from 'axios';
+import {Logger} from '../../types/Logger';
+import {Account} from '../../types/Account';
+
+export class IdamApi {
+
+  private readonly addUser: string = config.get('services.idam.addNewUserURL');
+
+  constructor(
+    private readonly axios: AxiosInstance,
+    private readonly logger: Logger
+  ) {}
+
+  public registerUser(account: Account, accessToken: string): Promise<Account>{
+    return this.axios
+      .post(`${this.addUser}`, account, {
+        headers: {
```

Comment:
I think if we refer to src/main/modules/oidc/index.ts server.use route, we don't need to specify the header for the authorization token here, as we can do something akin to: 

    server.use((req: AuthedRequest, res: Response, next: NextFunction) => {
      if (req.session.user) {
        req.scope = req.app.locals.container.createScope();
        req.scope.register({
          axios: asValue(Axios.create({
            baseURL: config.get('services.api.url'),
            headers: {
              Authorization: 'Bearer ' + req.session.user.id_token
            }
          })),
          api: asClass(FactApi)
        });

        res.locals.isLoggedIn = true;
        res.locals.isSuperAdmin = req.session.user.jwt.roles.includes('fact-super-admin');

        return next();
      }

      if (req.xhr) {
        res.status(302).send({ url: '/login' });
      } else {
        res.redirect('/login');
      }
    });